item.rb

```diff
@@ -0,0 +1,34 @@
+require 'date'
+
+class Item
+  attr_reader :id, :genre, :author, :source, :label, :publish_date, :archived
+
+  def initialize(id = Random.rand(1..1000))
+    @id = id
+    @genre = nil
+    @author = nil
+    @source = nil
+    @label = nil
+    @publish_date = nil
+    @archived = false
+  end
+
+  def move_to_archive
+    @archived = true if can_be_archived?
+  end
+
+  private
+
+  def can_be_archived?
+    begin
+      publish_date = Date.parse(@publish_date)
+    rescue StandardError
+      publish_date = nil
+    end
+    if publish_date.nil?
+      false
+    else
+      (Date.today.year - publish_date.year) > 10
+    end
+  end
+end
```

Comment:
- [ ] In the item class, there is supposed to be a Genre, Author, Label method that the students did not implement and was not mentioned by the previous code reviewer. For an example, I will include one for you below:

```
def genre=(genre)
      @genre = genre
      genre.items.push(self) unless genre.items.include?(self)
    end
```